micrometer-test/src/main/java/io/micrometer/core/instrument/HttpClientTimingInstrumentationVerificationTests.java

```diff
@@ -49,6 +62,39 @@ public enum HttpMethod {
 
     }
 
+    /**
+     * Provide your client with instrumentation required for registering metrics.
+     * @return instrumented client
+     */
+    protected abstract CLIENT clientInstrumentedWithMetrics();
+
+    /**
+     * Provide your client with instrumentation required for registering metrics via
+     * {@link ObservationRegistry}.
+     * @return instrumented client or {@code null} if instrumentation with
+     * {@link Observation} is not supported
+     */
+    @Nullable
+    protected abstract CLIENT clientInstrumentedWithObservations();
+
+    /**
+     * Returns the instrumented client depending on the {@link TestType}.
+     * @return instrumented client with either {@link MeterRegistry} or
+     * {@link ObservationRegistry}
+     */
+    protected CLIENT instrumentedClient() {
```

Comment:
This works, but I wonder if users implementing this class need the instrumented client anywhere but in the `sendHttpRequest` method implementation. Maybe we just add the instrumented http client as an argument there, and then this method can be internal (private). WDYT?

Reply:
Great idea!